api/proxy/config/enabled_apis/enabled_apis.go

```diff
@@ -0,0 +1,172 @@
+package enabled_apis
+
+import (
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/Layr-Labs/eigenda/api/proxy/common"
+)
+
+// EnabledAPIs is a wrapper type of the APIs set enum and provides
+// encapsulation for the entrypoint processing to make enablement determination
+// through specific boolean accessors.
+type EnabledAPIs struct {
+	apis []API
+}
+
+func (e EnabledAPIs) Count() uint {
+	return uint(len(e.apis))
+}
+
+// ArbCustomDA ... Is Arbitrum Custom DA enabled?
+func (e EnabledAPIs) ArbCustomDA() bool {
+	return e.has(ArbCustomDAServer)
+}
+
+// RestALTDA ... Is the REST ALTDA server enabled?
+func (e EnabledAPIs) RestALTDA() bool {
+	return e.has(Admin) || e.has(OpGenericCommitment) ||
+		e.has(OpKeccakCommitment) || e.has(StandardCommitment)
+}
+
+// RestALTDAOPGeneric ... Is REST server enabled with op-generic commitment mode
+func (e EnabledAPIs) RestALTDAOPGeneric() bool {
+	return e.has(OpGenericCommitment)
+}
+
+// RestALTDAOPKeccak ... Is REST server enabled with op-keccak commitment mode
+func (e EnabledAPIs) RestALTDAOPKeccak() bool {
+	return e.has(OpKeccakCommitment)
+}
+
+// RestALTStandard ... Is REST server enabled with standard commitment mode
+func (e EnabledAPIs) RestALTStandard() bool {
+	return e.has(StandardCommitment)
+}
+
+// RestALTDAWithAdmin ... Is REST server enabled with admin mode
+func (e EnabledAPIs) RestALTDAWithAdmin() bool {
+	return e.has(Admin)
+}
+
+// Metrics ... Are metrics exposed on the metrics server?
+func (e EnabledAPIs) Metrics() bool {
+	return e.has(MetricsServer)
+}
+
+// Check ... Ensures that expression of the enabled API set is correct
+func (e EnabledAPIs) Check() error {
+	if e.Count() == 0 {
+		return fmt.Errorf("expected at least one \"apis.enabled\" value to be provided")
+	}
+
+	if common.ContainsDuplicates(e.apis) {
+		return fmt.Errorf("enabled apis contains duplicate: %+v", e.apis)
+	}
+
+	if e.Metrics() && (!e.RestALTDA() && !e.ArbCustomDA()) {
+		return fmt.Errorf("metrics cannot be enabled unless `arb` and/or `rest` also is")
+	}
+
+	if e.RestALTDAWithAdmin() &&
+		(!e.RestALTDAOPGeneric() && !e.RestALTDAOPKeccak() && !e.RestALTStandard()) {
+		return fmt.Errorf("admin mode for REST ALTDA server cannot be enabled without also " +
+			"setting one of `op-generic`, `op-keccak`, `standard`")
+	}
```

Comment:
here also the admin is not the main point. main point is one of the data routes needs to be enabled.

Reply:
agree with the first comment but why would admin exposed without not be the main check here for a proper api types expression for the REST ALT DA Server?

Reply:
I'm just saying the check should instead enforce one of op-generic, op-keccak, OR standard to be turned on. ALWAYS.
Then if we have that you can turn on/off admin API orthogonally. We don't need this check.